### ISSUE TITLE
Test L4Rule CRD for AG in KLAB2

### ISSUE THREAD

**Issue by StevenBarre** [bcgov/developer-experience#5494]
**Describe the issue**
Test applying custom network access policies to routes via L4Rule CRDs in KLAB2

**What is the Value/Impact?**
More secure routes in Emerald

**What is the plan? How will this get completed?**
Work with AG to stand up a test route in KLAB2
Add L4Rule to restrict access
Test access

**Identify any dependencies**
None

**Definition of done**
Docs updated with examples

**Comment by StevenBarre** [comment]
There is no way in the web gui to create a Network Security Policy.
Tested copying the NSP from one Virtual Server to another with the L4Rule object.

ToDo:
- Test if the rules are actually enforced

**Comment by wmhutchison** [comment]
The ongoing email thread with AG made it sound like they are going the exploratory route of upgrading security class to address their issues instead of what was worked out for this ticket. This might change next week when people are back from holiday vacations.

**Comment by StevenBarre** [comment]
Testing Multiple AKO instances https://techdocs.broadcom.com/us/en/vmware-security-load-balancing/avi-load-balancer/avi-kubernetes-operator/1-12/avi-kubernetes-operator/multiple-ako-instances-in-a-cluster.html

This would allow pushing all AG routes to their AVI tenant for them to see/manage directly themselves.

**Comment by StevenBarre** [comment]
Blocked waiting on feedback from Broadcom

**Comment by StevenBarre** [comment]
v1.13.1 has been released and supposedly fixes the issues. Will test next sprint.

**Comment by StevenBarre** [comment]
Tested v1.13.1 . Some parts work, but the two pods fight over the global objects like AviInfraSettings and cause issues. Emailed support some fresh logs.

**Comment by StevenBarre** [comment]
We've now got one AKO instance managing two tenants. We can create new VIP networks at a global scope available to all tenants.

To switch a namespace out of the default tenant add this annotation to the namespace `ako.vmware.com/tenant-name:` then restart AKO.

Since the default VIP is not available in all tenants, we'll need to create additional AVIInfraSettings objects for the new networks and then annotate all routes in the tenants NS with one of those new "dataclasses".

Then the route will be created in the secondary tenant, where they can manage ACLs and WAF policies via the AVI web console.

**Comment by StevenBarre** [comment]
Had a call with AG and they like our plan. Next up is to make it operational in Emerald and document it.

**Comment by StevenBarre** [comment]
https://citz-hsb.atlassian.net/browse/SDNP-257 is with Thom Bagnall for review.

**Comment by StevenBarre** [comment]
Closing for now, will make a new ticket when we are approved to move forward with implementation.
